README.md

```diff
@@ -139,26 +139,129 @@ Dedicated servers hosted on Steam are usually required to be running the *latest
+```
+
+* `-t` for a pseudo-TTY is mandatory; servers may not run correctly without it
+* `-i` for STDIN for interactive use of the game console
+* `-d` for running the container in detached mode
+
+#### Attaching
+
+If the game process is running as `PID 1` and STDIN is enabled for the container, the game's console can be accessed via:
+
+```shell
+docker attach containername
+```
+
+#### Debugging
+
+To debug a container or its files:
+
+```shell
+# To enter into a container
+docker exec -it containername bash
+
+# To issue detached command(s)
+docker exec containername ps aux                                     # Single, simple command
+docker exec containername bash -c 'printenv && ls -al && ps aux'     # Multiple or advanced commands
+```
+
+## Important considerations
+
+Due to the variety of `SRCDS` and `HLDS` games that can be hosted and the various ways each of the games can and/or have to be hosted, the images built using this project are kept to be as generic as possible. The following are important considerations concerning the images provided by the project.
+
+### Entrypoint script
+
+The game images **do not** include an entrypoint script.
+
+While the conventional `entrypoint.sh` could have been included, having so also takes away flexibility for how the images can be used. Operators wishing to utilize their own entrypoint scripts would have to include removal of pre-existing ones as part of their build or init process, which likely adds unnecessary confusion. It is also unlikely that a generic entrypoint script would be adequate given the various ways server operators could implement the container's initialization process for the game server.
+
+This brings us to the next but a much related consideration.
+
+### Environment variables
+
+The game images **do not** include support for mapped environment variables.
+
+Docker images are often designed with environment variables that are typically mapped onto arguments which are generated within the included entrypoint script as part of the container's initialization process that are then used for invoking the core binary for which the image is built.
+
+While support for them could be made for the provided game images, having so assumes not only the presence of an entrypoint script, but environment variables with names that would likely serve to add only more confusion to server operators due to the sheer number parameters (e.g. `-usercon`) and Cvars (e.g. `+port`) that too differs depending on the game engine (`SRCDS`, `HLDS`) and game. Moreover, the command line of game servers can greatly vary depending on how the servers are hosted, making it difficult to say which parameters or Cvars should be considered mandatory and so be mapped and configurable via environment variables. Such support introduces an unnecessary layer of abstraction that adds more problems than it solves.
```

Comment:
This is how I will write it. Feel free to insert it anywhere that makes the matter clearer so people won't open unnecessary issues on this.

"While a gameserver might be configured with parameters (i.e. flags that begin with `-`, e.g. `-console`. See [srcds parameters](https://developer.valvesoftware.com/wiki/Command_Line_Options#Command-line_parameters_5) and [`hlds` parameters ](https://developer.valvesoftware.com/wiki/Command_Line_Options#Command-line_parameters_2)), a gameserver might also be configured with `cvars` (i.e. flags that begin with `+`, e.g. `+sv_lan 0`. See [srcds Console variables](https://developer.valvesoftware.com/wiki/Command_Line_Options#Console_variables) and [hlds Console variables](https://developer.valvesoftware.com/wiki/Command_Line_Options#Useful_console_variables)). Although there are many `cvars` common across all `srcds` games as well as across `hlds` games, there are also `cvars` that may be game-specific, for instance. `left4dead` or `left4dead2` which have several hundred of game-specific `cvars`. There may also be custom `cvars` created by game mod plugins e.g. `garrysmod`, `amxmodx`, `sourcemod`. Because of the malleable nature of how cvars are used, it does not make sense to map them directly to environment variables for several reasons: First, it introduces an unnecessary layer of abstraction which all users have to learn, on top of the difficulty in learning `cvars` alone; Second, a single change in single envvar-cvar mapping will require a rebuild of docker image new `docker-entrypoint.sh` script, introducing a lot of unnecessary builds; Third, the very `docker-entrypoint.sh` script with envvar-cvar mapping now must be versioned, introducing another burden than just keeping docker images updated. 

Therefore, to keep things simple, there is no support for environment variables mapping onto parameters and `cvars`.